DataCollection.py

```diff
@@ -1,45 +1,92 @@
 import streamlit as st
 import pandas as pd
 import openpyxl
+from io import BytesIO
 
-# Fungsi untuk memeriksa apakah file Excel memiliki sel yang digabung (merged)
-def is_excel_merged(file_path):
+# Function to preprocess the data based on user selections
+def preprocess_data(data, selected_columns, preprocessing_method):
+    if preprocessing_method == "Drop":
+        data = data.dropna(subset=selected_columns)
+    elif preprocessing_method == "Imputation":
+        for column in selected_columns:
+            if data[column].dtype == 'object':
+                # Impute with mode based on the most frequent value
+                mode_value = data[column].mode()[0]
+                data[column].fillna(mode_value, inplace=True)
+            elif data[column].dtype == 'float64':
```

Comment:
kalau tipe datanya int64 gimana? or any other numerical data type?

can we check whether the data type is numerical instead of checking for a certain data type?